docs/guide/database-insert-update.md

```diff
@@ -47,26 +53,34 @@ For example, if we had a dog adoption website we may want to update `dog` pieces
 **What is happening here?**
 
 ```javascript
-async registerInterest(dogId) {
-  const req = apos.task.getReq({ role: 'editor' });
-  const dogDocument = await self.find(req, { _id: dogId }).toObject();
+async updateDogStatus(req, dogId, status) {
+  const dogDocument = await self.find(req, {
+    dogId
+  }).toObject();
   ...
 }
 ```
 
-Our method received the `_id` of the document associated with this dog. We then generate a `req` object using `apos.task.getReq()`. While it is best to use a "natural" request object, if the contact form submission was from someone not logged in then the `req` will not have permission to update documents. We would want to make sure to guard against spam submissions elsewhere.
+Our method received the active request object, the API's identifier for a dog, and the dog's availability status (probably a string value). We then request the *full* database document. See the [database querying guide](/guide/database-queries.md) for more on this.
+
+```javascript
+if (!dogDocument) {
+  return null;
+};
+```
 
-We then request the *full* database document. See the [database querying guide](/guide/database-queries.md) for more on this.
+If the identified dog isn't in our database yet, we would return a `null` value and the update task's function could insert the dog instead.
 
 ```javascript
-dogDocument.receivedInterest++;
+// We update the `status` property.
+dogDocument.status = status;
 
 const updateResult = await self.update(req, dogDocument);
```

Comment:
This does read better, but we should mention that:

* This will only work if `req` is for a user with permission to at least contribute to dogs. You can pass `{ permissions: false }` as the options argument to `update` if you want to allow the public or a guest user to update the status of a dog in this way.
* The update method will only modify the current draft, unless the piece type has the`localize: false` option or the `autopublish: true` option. There is a publish method available, if you want to follow up with that in a particular case. Also a submit method, if you want to treat it as a contribution so someone with permission can confirm it.

Just don't want people to turn right around and say it "doesn't work."